triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -67,8 +71,62 @@ object TriggerServiceFixture {
     val host = InetAddress.getLoopbackAddress
     val isWindows: Boolean = sys.props("os.name").toLowerCase.contains("windows")
 
-    // Launch a toxiproxy instance. Wait on it to be ready to accept
-    // connections.
+    // Set up an authentication service if enabled
+    val authServiceAdminLedger: Future[Option[(SandboxServer, Port)]] =
+      if (!auth) Future(None)
+      else {
+        val adminLedgerId = LedgerId("admin-ledger")
+        for {
+          ledger <- Future(
+            new SandboxServer(
+              SandboxServer.defaultConfig.copy(
+                port = Port.Dynamic,
+                ledgerIdMode = LedgerIdMode.Static(adminLedgerId),
+              ),
+              mat))
+          ledgerPort <- ledger.portF
+        } yield Some((ledger, ledgerPort))
+      }
+
+    val authServiceBinaryLoc: String = {
+      val extension = if (isWindows) ".exe" else ""
+      BazelRunfiles.rlocation("triggers/service/ref-ledger-authentication-binary" + extension)
+    }
+
+    val authServiceInstanceF: Future[Option[(Process, Uri)]] =
```

Comment:
Shouldn’t this be using the fixture?

Reply:
Yeah, I'd like to use the `AuthServiceFixture` I wrote before but didn't see immediately how to compose these fixtures. Will try to reuse the code more cleanly once it's working